### PR TITLE
WIP: Feature/coreos

### PR DESCRIPTION
**Pull request by johnt337**
Creating PR to facilitate discussion around the introduction of CoreOS support into KOPS.
@chrislovecnm @kris-nova

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/854)

### CODE REVIEW

File: README.PR.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+My initials thoughts
+
+1. docker/rkt are bundled by default with CoreOS so its a matter of what configs we need to apply to satisfy k8s. Patterns for handling this the "CoreOS way" are already documented. Typically handled as systemd drop-ins via cloud-init
+2. Items that are installed via apt-get (or into the /usr partition) would require work arounds to deal with a read only partition and no installer (i.e. bundled via container)
+3. Leveraging user-data will maximize configuration and immutability of the deployment
+4. Size limits of cloud-init would influence approach
+    - 16K limit - http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+    - Embedding all the k8s manifests, certs, and add-ons directly in the user-data may not feasible if they exceed this limit
+    - They might need to be externalized and retrieved during boot time (would that go in KOPS_STATE_STORE?)
+5. Systemd units hopefully would be portable (need to confirm)
```
Comment (chancez) [inline README.PR.md]: They likely aren't, and most likely, a lot of them are unnecessary/aren't "the CoreOS way" (ex: only  kubelet, flannel, docker run as systemd units, the latter 2 are included in the OS). Additionally, I'd imagine unit names, dependencies are different.

File: README.PR.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+6. How much would you want to handle with nodeup versus straight cloud-init?
+7. Networking - what type of networking would you want to tackle first?
+8. Preference between cloud-init or ignition?
```
Comment (chancez) [inline README.PR.md]: Generally prefer ignition when possible.

**Comment by chrislovecnm** [comment]
@justinsb can you review this?

@chancez would it be worth while to meet on a zoom call about this

File: README.PR.md (hunk 2 above)
Comment (hsyed) [inline README.PR.md]: I would vote for an approach that reduces cloud init / ignition scripting to passing a handful of keys and starting a top level one shot unit. I use this approach for our clusters, we have the luxury of an EFS mount though.

The one shot unit could simply pull down a bootstrap container and start off the up process. 

I would really like to see the Coreos bootstrap be done In a model driven way with go code rendering the units from templates and starting them.

**Comment by chancez** [comment]
@chrislovecnm I unfortunately do not have a lot of free time for this, as this isn't my area of focus at CoreOS. I would be glad to review and provide guidance in my (limited) free time however.

**Comment by chrislovecnm** [comment]
Do we want to keep this open or get this into an issue?

**Comment by chrislovecnm** [comment]
I am going to close this out as we have work underway

**Pull request closed without merging**